DFC.ServiceTaxonomy.Taxonomies/Controllers/AdminController.cs

```diff
@@ -340,41 +352,19 @@ private JObject FindTaxonomyItem(JObject contentItem, string taxonomyItemId)
             return null;
         }
 
-        private ContentItem FindParentTaxonomyTerm(ContentItem termContentItem, ContentItem taxonomyContentItem)
-        {
-            List<ContentItem> terms = GetTerms(taxonomyContentItem);
-
-            if (terms == null)
-                return null;
-
-            if (terms.Any(x => x.ContentItemId == termContentItem.ContentItemId))
-                return taxonomyContentItem;
-
-            ContentItem result = null;
-
-            foreach (var term in terms)
-            {
-                result = FindParentTaxonomyTerm(termContentItem, term);
-
-                if (result != null)
-                {
-                    return result;
-                }
-            }
-
-            return null;
-        }
-
         private bool ValidateTaxonomyTerm(ContentItem parent, ContentItem term)
         {
-            List<ContentItem> terms = GetTerms(parent);
+            List<ContentItem> terms = TaxonomyHelpers.GetTerms(parent);
             return terms?.All(x => x.ContentItemId == term.ContentItemId || x.DisplayText != term.DisplayText) ?? true;
         }
 
-        private List<ContentItem> GetTerms(ContentItem contentItem)
+        private async Task<bool> ValidateTaxonomyTermURL(ContentItem term, ContentItem taxonomy)
         {
-            return contentItem.As<TaxonomyPart>()?.Terms ??
-                   contentItem.Content.Terms?.ToObject<List<ContentItem>>();
+            string url = TaxonomyHelpers.BuildTermUrl(term, taxonomy);
+            //TODO: check whether or not we only care about published pages, but I think we care about both
```

Comment:
Getting Latest will get a mix of published and preview.
I think it needs to separately check both published and preview.

Reply:
Ignore me, we're only catering for published taxonomies, so latest should be fine.